File: app/src/main/java/me/shadura/escposprint/L.java
package me.shadura.escposprint;

import android.support.annotation.Nullable;
import android.util.Log;

/**
 * Logging/crash reporting functions
 */
public class L {
    /**
     * Verbose log
     *
     * @param msg Log message
     */
    public static void v(String msg) {
        Log.v(EscPosPrintApp.LOG_TAG, msg);
    }

    /**
     * Info log
     *
     * @param msg Log message
     */
    public static void i(String msg) {
        Log.i(EscPosPrintApp.LOG_TAG, msg);
    }

    /**
     * Warning log
     *
     * @param msg Log message
     */
    public static void w(String msg) {
        Log.w(EscPosPrintApp.LOG_TAG, msg);
    }

    /**
     * Debug log
     *
     * @param msg Log message
     */
    public static void d(String msg) {
        Log.d(EscPosPrintApp.LOG_TAG, msg);
    }

    /**
     * Error log
     *
     * @param msg Log message
     */
    public static void e(String msg) {
        Log.e(EscPosPrintApp.LOG_TAG, msg);
    }

    /**
     * Error reporting
     *
     * @param msg Log message
     * @param t   Throwable to print, if not null
     */
    public static void e(String msg, @Nullable Throwable t) {
        e(msg);
        if (t != null) {
            e(t.getLocalizedMessage());
        }
    }
}

File: app/src/main/java/me/shadura/escposprint/L.kt
package me.shadura.escposprint

import com.hypertrack.hyperlog.HyperLog

/**
 * Logging/crash reporting functions
 */
object L {
    /**
     * Verbose log
     *
     * @param msg Log message
     */
    fun v(msg: String) {
        HyperLog.v(EscPosPrintApp.LOG_TAG, msg)
    }

    /**
     * Info log
     *
     * @param msg Log message
     */
    fun i(msg: String) {
        HyperLog.i(EscPosPrintApp.LOG_TAG, msg)
    }

    /**
     * Warning log
     *
     * @param msg Log message
     */
    fun w(msg: String) {
        HyperLog.w(EscPosPrintApp.LOG_TAG, msg)
    }

    /**
     * Debug log
     *
     * @param msg Log message
     */
    fun d(msg: String) {
        HyperLog.d(EscPosPrintApp.LOG_TAG, msg)
    }

    /**
     * Error log
     *
     * @param msg Log message
     */
    fun e(msg: String) {
        HyperLog.e(EscPosPrintApp.LOG_TAG, msg)
    }

    /**
     * Error reporting
     *
     * @param msg Log message
     * @param t   Throwable to print, if not null
     */
    fun e(msg: String, t: Throwable?) {
        e(msg)
        if (t != null) {
            e(t.localizedMessage ?: t.message ?: "<no message>")
        }
    }


}
